### DISCUSSION TITLE
Trying to include execute_params in a df for use in a call to quarto_render()

### DISCUSSION THREAD

**Discussion by ghost** [quarto-dev/quarto-cli#4323] [Q&A]
I am trying to take vars from a df and use them as params in a call to quarto_render:
```
library(tidyverse)
library(quarto)
one_person <- tibble(
  title=c("Rev"), 
  name=c("Albert")
)

test_tibble_2 <- tibble(
  input='index.qmd',
  output_format = "html",
  output_file = paste0("tibble2_", 
                       one_person$title, 
                       "_", 
                       one_person$name,
                       ".html"),
   execute_params = list(params=one_person$title)
)


pmap(
  test_tibble_2, 
  quarto_render
)



```
This approach works if I don't include execute_params (which was the point) but fails when I do.  My guess is that either  the 
one_person is not structured properly, or the execute_params is incorrectly specified. 

This is the failure I get; 

_Error in knit_params_get(input_lines, params) :    render params argument must be a named list Calls: .main -> execute -> <Anonymous> -> knit_params_get Execution halted Error in `pmap()`: ℹ In index: 1. Caused by error: ! System command 'quarto' failed Run `rlang::last_error()` to see where the error occurred.
--_

**Comment by mcanouil** [comment]
Could you properly format your post, please?
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>

And possibly change the category from "general" to "Q&A".

**Comment by ghost** [comment]
Re-written and posted to Q&A.

**Reply by ghost** [reply]
https://github.com/quarto-dev/quarto-cli/discussions/4327

**Reply by mcanouil** [reply]
There was no need to recreate a post, simply change the category using a browser with "desktop view/mode".

**Comment by mcanouil** [comment]
`execute_params` is not for metadata.
It is for parameters metadata, i.e., under `params` YAML key, see <https://quarto-dev.github.io/quarto-r/reference/quarto_render.html> and <https://quarto.org/docs/computations/parameters.html>.

The only way you can make it work is as follow.

- `file.qmd`
```qmd
---
title: '`r params[["title"]]`'
params:
  title: "Default"
---

Text
```

- `script.r`
```r
quarto_render("file.qmd", execute_params = list(title = "Replace the default title with this one"))
```

If you want to override directly the metadata (_e.g._, `lang`, etc., <https://github.com/quarto-dev/quarto-r/issues/49>), you actually can't with the current stat of the R package.
Note that there is a pending pull request, see <https://github.com/quarto-dev/quarto-r/pull/52>.
